gems/pending/spec/util/miq-file_spec.rb

```diff
@@ -0,0 +1,38 @@
+require 'util/extensions/miq-file'
+
+describe 'MIQFile' do
+  context 'path_to_uri' do
+    let(:hostname) { MiqSockUtil.getFullyQualifiedDomainName }
+
+    it 'is defined and returns a string' do
+      expect(File).to respond_to(:path_to_uri)
+      expect(File.path_to_uri('foo')).to be_kind_of(String)
+    end
+
+    it 'returns the expected result' do
+      expect(File.path_to_uri('foo')).to eql("file://#{hostname}/foo")
+      expect(File.path_to_uri('foo/bar')).to eql("file://#{hostname}/foo/bar")
+      expect(File.path_to_uri('foo/bar-stuff')).to eql("file://#{hostname}/foo/bar-stuff")
+      expect(File.path_to_uri('foo/C:/bar')).to eql("file://#{hostname}/foo/C:/bar")
+      expect(File.path_to_uri('foo/[bar]')).to eql("file://#{hostname}/foo/%5Bbar%5D")
+    end
+
+    it 'accepts an optional hostname and returns the expected result' do
+      hostname = 'dell-r410-01.cloudformswin.lab.redhat.com'
```

Comment:
We generally try not to use internal names, maybe use `dell-r410-01.manageiqwin.lab.example.com`